fdbcli/fdbcli.actor.cpp

```diff
@@ -989,6 +989,80 @@ void printStatus(StatusObjectReader statusObj, StatusClient::StatusLevel level,
 
 				if (statusObjConfig.get("log_routers", intVal))
 					outputString += format("\n  Desired Log Routers    - %d", intVal);
+
+				outputString += "\n  Usable Regions         - ";
+				if (statusObjConfig.get("usable_regions", intVal)) {
+					outputString += std::to_string(intVal);
+				} else {
+					outputString += "unknown";
+				}
+
+				Optional<std::string> activePrimaryDC;
+
+				if (statusObjConfig.has("active_primary_dc")) {
+					activePrimaryDC = statusObjConfig["active_primary_dc"].get_str();
+				}
+
+				StatusArray regions;
+				if (statusObjConfig.has("regions")) {
+					outputString += "\n  Regions: ";
+					regions = statusObjConfig["regions"].get_array();
+					bool isFirstRegion = true;
+					bool isPrimary = false;
+					std::vector<std::string> regionSatelliteDCs;
+					std::string regionDC;
+					for (StatusObjectReader region : regions) {
+						// Assumes the `regions` are already sorted by priority in descending order
+						for (StatusObjectReader dc : region["datacenters"].get_array()) {
+							if (!dc.has("satellite")) {
+								regionDC = dc["id"].get_str();
+								if (activePrimaryDC.present() && dc["id"].get_str() == activePrimaryDC.get()) {
+									isPrimary = true;
+								} else if (!activePrimaryDC.present() && isFirstRegion) {
```

Comment:
Does `activePrimaryDC` not being present in the database imply that the highest priority region is the primary, or is this written to handle the case where we couldn't read the key for some reason?

If is this is to handle the error case, I think we should fallback to printing something generic, like "Region" instead of "Primary/Remote". If we also add a message to status in this case, then we should also get a nice message at the top of the output to say something like "Unable to determine primary datacenter".

Reply:
> Does activePrimaryDC not being present in the database imply that the highest priority region is the primary, or is this written to handle the case where we couldn't read the key for some reason?

The code is intended to handle both cases. If `activePrimaryDC` cannot be determined(failed to read from DB or it's not set in DB), we use the region with highest priority as the primary. @etschannen is this a correct assumption?

>If is this is to handle the error case, I think we should fallback to printing something generic, like "Region" instead of "Primary/Remote". If we also add a message to status in this case, then we should also get a nice message at the top of the output to say something like "Unable to determine primary datacenter".

Yeah good idea. Will do this.

Reply:
If the key can't be read, then I don't think we can assume anything. If it can be read and isn't set, then we should a) determine if that's possible, and b) understand what that means. For example, if the key is allowed to be unset and implies something about which datacenter is the primary, then we should be sure to encode that logic. If it being unset is either not supposed to happen or can't tell us anything about what the primary is, then we'll need to fall back to reporting that we don't know the answer.

Reply:
Done. Now if primary datacenter is missing it will print a warning and fall back to use the generic `Region` output. If it successfully got the active primary datacenter, it will do the `Primary` and `Remote` output. I confirmed with @etschannen that the primary datacenter should NOT be absent.